components/server/ee/src/user/eligibility-service.ts

```diff
@@ -314,7 +296,7 @@ export class EligibilityService {
         if (start === undefined) {
             return undefined;
         }
-        if (await this.subscriptionService.hasActiveAndNotYetCancelledPaidSubscription(user.id, date)) {
+        if (await this.subscriptionService.hasActivePaidSubscription(user.id, date)) {
```

Comment:
This is an edge case, I'm sure we can handle the impact. We dearly need (more) tests including reasoning for this kind of policy changes. :see_no_evil: